### PR TITLE
DEVX-77 migrate CategoryBatchValidator

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

JIRA: https://commercetools.atlassian.net/browse/DEVX-77

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  public CategoryBatchValidator(
+      @Nonnull final CategorySyncOptions syncOptions,
+      @Nonnull final CategorySyncStatistics syncStatistics) {
+
+    super(syncOptions, syncStatistics);
+  }
+
+  /**
+   * Given the {@link java.util.List}&lt;{@link io.sphere.sdk.categories.CategoryDraft}&gt; of
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java:33]:
```suggestion
   * Given the {@link java.util.List}&lt;{@link CategoryDraft}&gt; of
```

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+   * drafts this method attempts to validate drafts and collect referenced keys from the draft and
+   * return an {@link org.apache.commons.lang3.tuple.ImmutablePair}&lt;{@link
+   * java.util.Set}&lt;{@link io.sphere.sdk.categories.CategoryDraft}&gt;,{@link
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java:36]:
```suggestion
   * java.util.Set}&lt;{@link CategoryDraft}&gt;,{@link
```

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+   * <p>A valid category draft is one which satisfies the following conditions:
+   *
+   * <ol>
+   *   <li>It is not null
+   *   <li>It has a key which is not blank (null/empty)
+   * </ol>
+   *
+   * @param categoryDrafts the category drafts to validate and collect referenced keys.
+   * @return {@link org.apache.commons.lang3.tuple.ImmutablePair}&lt;{@link java.util.Set}&lt;{@link
+   *     io.sphere.sdk.categories.CategoryDraft}&gt;,{@link
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java:49]:
```suggestion
   *     CategoryDraft}&gt;,{@link
```

**Review by salander85 — APPROVED** [review]

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java (hunk 3 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java]: https://github.com/commercetools/commercetools-sync-java/pull/969/commits/e837176364c04cd7e405bcf41d52add9a387f28a

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java (hunk 2 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java]: https://github.com/commercetools/commercetools-sync-java/pull/969/commits/e837176364c04cd7e405bcf41d52add9a387f28a

File: src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java (hunk 1 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/categories/helpers/CategoryBatchValidator.java]: https://github.com/commercetools/commercetools-sync-java/pull/969/commits/e837176364c04cd7e405bcf41d52add9a387f28a

**Pull request merged**